nncf/common/accuracy_aware_training/training_loop.py

```diff
@@ -127,9 +134,26 @@ def _run_early_exit_training_loop(self, model):
 
             self.runner.update_learning_rate()
 
-        self.runner.load_best_checkpoint(model)
+        self._finish_early_exit_training_loop(model)
         return model
 
+    def print_maximal_degradation_warning(self, logger):
```

Comment:
It looks like a user code. I would suggest returning the final stats and let the user decide how they want to proceed.

Reply:
Done. I replaced call of this method with printing of final statistics after the loop finish. The idea is to demonstrate the user that there is such property as `final_statistics`.